cdc/kv/client.go

```diff
@@ -988,6 +988,11 @@ func (s *eventFeedSession) handleError(ctx context.Context, errInfo regionErrorI
 				zap.String("rpcCtx", errInfo.rpcCtx.String()),
 				zap.Stringer("error", compatibility))
 			return cerror.ErrVersionIncompatible.GenWithStackByArgs(compatibility)
+		} else if mismatch := innerErr.GetClusterIdMismatch(); mismatch != nil {
```

Comment:
does this case need covered by test ?

Reply:
We can add a unit test similar to `TestCompatibilityWithSameConn`

Reply:
Added. Thanks for your help!